core/src/main/java/io/grpc/InternalMetadata.java

```diff
@@ -46,27 +46,11 @@
   /**
    * A specialized plain ASCII marshaller. Both input and output are assumed to be valid header
    * ASCII.
+   *
+   * <p>Extended here to break the dependency.
    */
   @Internal
-  public interface TrustedAsciiMarshaller<T> {
-    /**
-     * Serialize a metadata value to a ASCII string that contains only the characters listed in the
-     * class comment of {@link io.grpc.Metadata.AsciiMarshaller}. Otherwise the output may be
-     * considered invalid and discarded by the transport, or the call may fail.
-     *
-     * @param value to serialize
-     * @return serialized version of value, or null if value cannot be transmitted.
-     */
-    byte[] toAsciiString(T value);
-
-    /**
-     * Parse a serialized metadata value from an ASCII string.
-     *
-     * @param serialized value of metadata to parse
-     * @return a parsed instance of type T
-     */
-    T parseAsciiString(byte[] serialized);
-  }
+  public interface TrustedAsciiMarshaller<T> extends Metadata.TrustedAsciiMarshaller<T> {}
```

Comment:
`Metadata.TrustedAsciiMarshalleris package private.  This is the public accessor.  It needs to be in this file so that internal things can access it.  It is easier to restrict access to this file.